client/src/Service/File/S3FileUploader.php

```diff
@@ -53,6 +54,7 @@ private function getFileBodyAndFileName(UploadedFile $file): array
 
         $fileName = $this->fileNameFixer->addMissingFileExtension($file, $body);
         $fileName = $this->fileNameFixer->removeWhiteSpaceBeforeFileExtension($fileName);
```

Comment:
`removeWhiteSpaceBeforeFileExtension` is now potentially redundant given your new function replaces it with underscores.

Reply:
Yeah we were debating between ourselves on this one and decided the resulting filename readability gains of potentially having a bunch of `_` at the end of the filename before the extension was worth leaving this in.